### PR TITLE
Changed the timezone used for saving meetups and scheduling reminders

### LINKED ISSUE #41: Change timezone used for saving meetups and scheduling reminders
## Description
Currently the meetup time and reminder schedules are stored in `Local` timezone of the host machine. 
This creates issues when the bot is deployed on some cloud platform because the `Local` time for the host machine may be different.
## Changes to be made
In [`meetup.go`](https://github.com/osdc/bots/blob/master/telegram-bot/meetup.go), replace instances where `Local` time is used, to use Indian time instead.

One such instance:-
https://github.com/osdc/bots/blob/e3920af23da1d8bd3cf825a9152e055c19b57b29/telegram-bot/meetup.go#L98

## First time contributors
**Hi, If it is the first time that you contribute to the bot:**

Write a comment in this issue thread to let other possible contributors know that you are working on this bug. For eg : `Hey all, I would like to work on this issue.`, follow [Contributing to the bot 🔥✨](https://github.com/osdc/bots/blob/master/CONTRIBUTING.md) and feel free to ask anything related to this issue ✌️

### PR DESCRIPTION
**Pull request by starkah**
Fixes #41

### CODE REVIEW

File: telegram-bot/meetup.go [hunk 1]
```diff
@@ -28,10 +28,10 @@ type group struct {
 }
 type meetuplist struct {
 	Name  string `json:"name"`
-	Venue venue  `json: "venue"`
+	Venue  venue `json:"venue"`
```
Comment (daemon1024) [inline telegram-bot/meetup.go]: I dont think this change was intentional, kindly revert it back.

File: telegram-bot/meetup.go [hunk 2]
```diff
@@ -28,10 +28,10 @@ type group struct {
...
 	Date  string `json:"local_date"`
 	Time  string `json:"local_time"`
-	Group group  `json: "group"`
+	Group  group `json:"group"`
```
Comment (daemon1024) [inline telegram-bot/meetup.go]: same here

File: telegram-bot/meetup.go [hunk 3]
```diff
@@ -93,9 +93,13 @@ func addmeetup(ID int64, msgtext string, client mongo.Client) {
 		if err != nil {
 			log.Fatalln(err)
 		}
+		location, err := time.LoadLocation("Asia/Kolkata")
+		if err != nil {
+			panic(err)
```
Comment (daemon1024) [inline telegram-bot/meetup.go]: I don't think panic is preferred. Instead `log.Print(err)` and send a bot message. We are planning to do it in #43 but as well as start here only :).

cc @sidntrivedi012

**Review by daemon1024 — CHANGES_REQUESTED** [review]
@starkah Thanks for the PR 😄 . Requested some changes. Rest LGTM

@sidntrivedi012 @DelusionalOptimist What do you think?

**Review by DelusionalOptimist — COMMENTED** [review]
I changed the timezone of my system  to UTC to test the changes. The reminder works fine. Great job :+1: .

Just one thing, when sending a reminder the bot still uses its local timezone. So you'll need to modify `reminder` and `nextmeetup` functions too.    
starkah_bot_test

**Comment by starkah** [comment]
@daemon1024 @DelusionalOptimist thanks for pointing out the necessary changes. I'm on it.

**Comment by starkah** [comment]
@DelusionalOptimist Wouldn't it be more easier for the user if the reminder was in it's local time zone?
I suggest that in the first line of the reminder we could change it from "MEETUP REMINDER!" to " REMINDER - MEETUP IN 2 HOURS" or something similar. What do you think?

> I changed the timezone of my system to UTC to test the changes. The reminder works fine. Great job +1 .
> 
> Just one thing, when sending a reminder the bot still uses its local timezone. So you'll need to modify `reminder` and `nextmeetup` functions too.
> starkah_bot_test

**Comment by daemon1024** [comment]
@starkah Note that the users are the members of OSDC and IST would be the local timezone for em :) But the bot can be deployed anywhere and the timezone of the developer's local system or any environment shouldn't be a dependency.

**Comment by starkah** [comment]
@daemon1024 Makes sense. I'll try to do that.

File: telegram-bot/meetup.go [hunk 4]
```diff
@@ -150,20 +160,22 @@ func reminder(ID int64, client mongo.Client, s1 *gocron.Scheduler) {
...
+			"\t" + data.Date.In(location).Format("15:04") + "\n" +
 			"Venue -" + "\t" + data.Venue
-		if !time.Now().Local().Before(data.Date) {
-			//stops the scheduler when meetup is done
-			s1.Clear()
-		} else {
-			bot.Send(tbot.NewMessage(ID, nxtMeetupData))
-		}
+		bot.Send(tbot.NewMessage(ID, nxtMeetupData))
+		//stops the scheduler when meetup is done
```
Comment (daemon1024) [inline telegram-bot/meetup.go]:
```suggestion
		//stops the scheduler when reminder is sent
```

File: telegram-bot/meetup.go [hunk 5]
```diff
@@ -150,20 +160,22 @@ func reminder(ID int64, client mongo.Client, s1 *gocron.Scheduler) {
 	if err != nil {
 		log.Fatal(err)
 	}
+	location, err := time.LoadLocation("Asia/Kolkata")
+	if err != nil {
+		log.Print(err)
+		bot.Send(tbot.NewMessage(ID, "Couldn't detect the timezone, please try again"))
```
Comment (daemon1024) [inline telegram-bot/meetup.go]:
```suggestion
		bot.Send(tbot.NewMessage(ID, "Couldn't detect the timezone while processing reminder, please try again"))
```

File: telegram-bot/meetup.go [hunk 6]
```diff
@@ -134,12 +139,17 @@ func nextmeetup(ID int64, client mongo.Client) {
...
```
Comment (daemon1024) [inline telegram-bot/meetup.go]:
```suggestion
		bot.Send(tbot.NewMessage(ID, "Couldn't detect the timezone while saving meetup, please try again"))
```

**Review by daemon1024 — CHANGES_REQUESTED** [review]
Some small nits rest LGTM.

**Review by daemon1024 — APPROVED** [review]

**Review by DelusionalOptimist — APPROVED** [review]
LGTM. Great work @starkah. :rocket:

**Pull request merged**